canonical/ubuntu-advantage-client
# daemon: auto-attach when pro license added on gcp

## Notes:
The license-check commit is exactly the same as the last one we merged

The daemon commit is significantly different, but the details of fetching the license are the same.

The biggest thing here (besides making sure the daemon works for auto-upgrade-to-pro) is making sure the daemon doesn't even run on non-gcpgeneric platforms.

TODO:
- [x] make sure manual license-check test scripts still work and put them in the right folder
- [x] fix the couple failing integration tests
- [x] test to confirm ConditionPathExists is sufficient to work with the cloud-id-${cloud-id} cloud-init feature

## Proposed Commit Message


```
daemon: auto-attach when pro license added on gcp
```
```
license-check: remove timer-based license-check
```

## Test Steps

Run new integration tests. Maybe not every combination of series/platform, but run a sample of series and platforms to make sure they pass.
Run the manual test scripts
- the `gcp_` scripts require two arguments: (1) a deb of ua from this PR and (2) a deb of cloud-init that includes the changes in https://github.com/canonical/cloud-init/pull/1244

## Desired commit type


## Checklist
